src/common_subset.rs

```diff
@@ -173,22 +225,19 @@ impl<NodeUid: Clone + Debug + Display + Eq + Hash + Ord> CommonSubset<NodeUid> {
             }
         }
 
-        if let Ok((output, mut outgoing)) = result {
-            // Process Agreement outputs.
-            if let Some(b) = output {
-                outgoing.append(&mut self.on_agreement_result(proposer_id, b)?);
-            }
+        let (output, mut outgoing) = result?;
 
-            // Check whether Agreement has completed.
-            Ok((
-                self.try_agreement_completion(),
-                outgoing.into_iter().map(Output::Agreement).collect(),
-            ))
-        } else {
-            // error
-            result
-                .map(|(_, messages)| (None, messages.into_iter().map(Output::Agreement).collect()))
+        // Process Agreement outputs.
+        if let Some(b) = output {
+            outgoing.append(&mut self.on_agreement_result(proposer_id, b)?);
         }
+
+        // Check whether Agreement has completed.
+        let into_msg = |a_msg| Target::All.message(Message::Agreement(proposer_id.clone(), a_msg));
```

Comment:
There are no `Target::Node` Agreement messages. Annotating the message may fit into the common message delivery system on one hand, but on the other, the messaging system may take into account that any Agreement message is a multicast.